src/app/static/src/app/components/genericChart/GenericChartTable.vue

```diff
@@ -72,30 +69,80 @@
             columnsWithHierarchy() {
                 return this.tableConfig.columns.filter(column => column.data.hierarchyColumn);
             },
+            hierarchyColumnPaths() {
+               // Generate a dictionary of dictionaries allowing hierarchy label lookup for all  hierarchy columns (i.e. area)
+               // to save the hierarchy string for each option. This traverses the entire hierarchy once only, which is
+               // quicker than calling findPath on each row value.
+               const addPathsToDict = (option: NestedFilterOption, parentPath: string,  dict: Dict<string>) => {
+                   //Add the path for this item...
+                   dict[option.id] = parentPath;
+                   //...then adds paths for any children
+                   const path = parentPath + (parentPath.length ? "/" : "") + option.label;
+                   if (option.children) {
+                       option.children.forEach((child) => addPathsToDict(child as NestedFilterOption, path, dict));
+                   }
+               };
+
+               const result = {} as Dict<Dict<string>>;
+               this.columnsWithHierarchy.forEach((columnConfig) => {
+                   const columnId = columnConfig.data.hierarchyColumn!;
+                   const pathsDict = {};
+                   const column = this.columns.find((column) => column.id == columnId)!;
+                   // Deal with both array or object as top-level values type
+                   const values = Array.isArray(column.values) ? column.values : [column.values];
+                   values.forEach((value) => addPathsToDict(value, "", pathsDict));
+                   result[columnId] = pathsDict;
+               });
+               return result;
+            },
             labelledData() {
                 // Build a dictionary of the columns defined in the fetched dataset, which include all values and
                 // associated labels
                 const columnsDict = this.columns.reduce((dict, column) => ({...dict, [column.id]: column}), {} as Dict<GenericChartColumn>);
-                return this.filteredData.map(row => {
+
+                // Convert each label column's filter options to a dictionary of id to labels so we do not need to find values'
```

Comment:
Have you bench-marked this change?

Reply:
Yes, it made a small difference ~20ms - maybe not worth it for the increased obfuscation?

Reply:
Hmm, this suprises me actually - array lookup is usually quicker when number of items is small. Might be worth just double checking as like you say, this is a bit hard to read! But 20ms is non trivial so if that's correct then I'm in favour of keeping it.

Reply:
You're completely right, I tried again with the latest version of everything and actually this only makes a marginal difference ~1ms or so. I think I must have run the two versions before at different times with different loads on my machine. OK, I'll take this out, but see if I can do anything about the row copy.